src/Android/Avalonia.Android/Platform/AndroidNativeControlHostImpl.cs

```diff
@@ -0,0 +1,187 @@
+            }
+
+            public void Dispose()
+            {
+                if (_view != null && _attachedTo?._avaloniaView is ViewGroup parent)
+                {
+                    parent.RemoveView(_view);
+                }
+                _child = null;
+                _attachedTo = null;
+                _view?.Dispose();
+                _view = null;
+            }
+
+            public INativeControlHostImpl? AttachedTo
+            {
+                get => _attachedTo;
+                set
+                {
+                    CheckDisposed();
+
+                    var oldAttachedTo = _attachedTo;
+                    _attachedTo = (AndroidNativeControlHostImpl?)value;
+                    if (_attachedTo == null)
+                    {
+                        oldAttachedTo?._avaloniaView.RemoveView(_view);
+                    }
+                    else
+                    {
+                        _attachedTo._avaloniaView.AddView(_view);
+                    }
+                }
+            }
+
+            public bool IsCompatibleWith(INativeControlHostImpl host) => host is AndroidNativeControlHostImpl;
+
+            public void HideWithSize(Size size)
+            {
+                CheckDisposed();
+                _view.Visibility = ViewStates.Gone;
+            }
+
+            public void ShowInBounds(Rect bounds)
+            {
+                CheckDisposed();
+                if (_attachedTo == null)
+                    throw new InvalidOperationException("The control isn't currently attached to a toplevel");
+
+                bounds *= _attachedTo._avaloniaView.TopLevelImpl.RenderScaling;
+                _view.Visibility = ViewStates.Visible;
+                _view.LayoutParameters = new FrameLayout.LayoutParams((int)bounds.Width, (int)bounds.Height)
+                {
+                    LeftMargin = (int)bounds.X,
+                    TopMargin = (int)bounds.Y
+                };
+                _view.RequestLayout();
+            }
+        }
+    }
+
+    public class AndroidViewControlHandle : INativeControlHostDestroyableControlHandle, IDisposable
+    {
+        internal const string AndroidDescriptor = "JavaHandle";
+        
+        private View? _view;
+        private bool _disposeView;
+        
+        public AndroidViewControlHandle(View view, bool disposeView)
+        {
+            _view = view;
+            _disposeView = disposeView;
+        }
+        
+        public View View => _view ?? throw new ObjectDisposedException(nameof(AndroidViewControlHandle));
+        
+        public string HandleDescriptor => AndroidDescriptor;
+
+        IntPtr IPlatformHandle.Handle => _view?.Handle ?? default;
+
+        public void Destroy()
+        {
+            Dispose(true);
+        }
+        
+        void IDisposable.Dispose()
+        {
+            Dispose(true);
+        }
+        
+        ~AndroidViewControlHandle()
```

Comment:
Do we need to have a finalizer? Why view is owned by a _wrapper class_?

Reply:
Removed IDisposable pattern and simplified that wrapper class.